xbmc/platform/linux/CPUInfoLinux.cpp

```diff
@@ -359,8 +360,15 @@ float CCPUInfoLinux::GetCPUFrequency()
   if (m_freqPath.empty())
     return 0;
 
-  CSysfsPath path{m_freqPath};
-  return path.Get<float>() / 1000.0f;
+  try
+  {
+    CSysfsPath path{m_freqPath};
```

Comment:
We check `m_freqPath` above in the constructor. I think it would be better to do a similar check there so that it's not even used if it throws an error.

Reply:
Hmm, not sure about that. It sounds to me like CPUs can switch between online and offline at any time so doing a check during construction probably isn't beneficial.

Ideally we would search for a new online CPU every time a read fails but at the moment I'm not motivated to implement that ^^

Reply:
yea that might be overkill really for something that is purely for visual sugar.